### PR TITLE
chore(deps): update dependency gdata/vaas to v9

### PR DESCRIPTION
**Pull request by ata-no-one**
This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [gdata/vaas](https://togithub.com/GDATASoftwareAG/vaas) ([source](https://togithub.com/GDATASoftwareAG/vaas-php)) | require | major | `8.0.2` -> `9.0.1` |

---

> :warning: **Warning**
> 
> Some dependencies could not be looked up. Check the warning logs for more information.

---

### Release Notes

<details>
<summary>GDATASoftwareAG/vaas-php (gdata/vaas)</summary>

### [`v9.0.1`](https://togithub.com/GDATASoftwareAG/vaas-php/compare/v9.0.0...v9.0.1)

[Compare Source](https://togithub.com/GDATASoftwareAG/vaas-php/compare/v9.0.0...v9.0.1)

### [`v9.0.0`](https://togithub.com/GDATASoftwareAG/vaas-php/compare/v8.0.2...v9.0.0)

[Compare Source](https://togithub.com/GDATASoftwareAG/vaas-php/compare/v8.0.2...v9.0.0)

</details>

---

### Configuration

:date: **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

:recycle: **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Renovate Bot](https://togithub.com/renovatebot/renovate).

### CODE REVIEW

**Comment by ata-no-one** [comment]
### Edited/Blocked Notification

Renovate will not automatically rebase this PR, because it does not recognize the last commit author and assumes somebody else may have edited the PR.

You can manually request rebase by checking the rebase/retry box above.

 :warning: **Warning**: custom changes will be lost.

**Comment by unglaublicherdude** [comment]
So this is the minimal inversive version. If opcache is enabled and causes this problem we have to recomment the customer to blacklist the whole vendor/amphp folder from the opcache. [In this repo is only an example list](https://github.com/GDATASoftwareAG/nextcloud-gdata-antivirus/pull/98/files#diff-29ea034c16dbc78ba1059e9e1815d7d0b99846e6df4c7ed86614c7dfd38cf553), because it has to be full pathes.

You can generate this list by running this command in the vendor/amphp folder:

```bash
find ~+ -type f > blacklist.txt
```

**Comment by unglaublicherdude** [comment]
There might be one other Option. By using ini_set, but I currently don't know If that has an effect on the running script. If that works we should do it in our sdk

**Comment by unglaublicherdude** [comment]
And just to be clear. This is not an issue with nextcloud, this is an issue with async-php and opcache. So if we cannot merge this pr, we have to remove amphp from the php-sdk.

**Comment by unglaublicherdude** [comment]
ok, ini_set workaround does not work as it basically has no effect on the runtime

I guess, we have a different issue as https://github.com/php/php-src/issues/12249 because this was fixed a fiew php versions before. We are using 8.2.21 in the nextcloud-container.

**Comment by unglaublicherdude** [comment]
I opened this [issue on the revolt repository](https://github.com/revoltphp/event-loop/issues/95).

I opened this [issue on the amphp-httpclient repository](https://github.com/amphp/http-client/issues/365).

**Comment by unglaublicherdude** [comment]
Here 2 of the coredumps for the segmentation faults

[core-apache2-11-33-33-118-1723720563.zip](https://github.com/user-attachments/files/16624692/core-apache2-11-33-33-118-1723720563.zip)
[core-apache2-11-33-33-119-1723720563.zip](https://github.com/user-attachments/files/16624694/core-apache2-11-33-33-119-1723720563.zip)

**Comment by unglaublicherdude** [comment]
@lennartdohmann please look over this PR again. It is kinda mergable now. I would feel better if the amp/file package would have been already released with the fix resulting from our issue but I guess we have No influence on that.

But we should also put that fix into our sdk. Interestingly you can Install branches from packages by using dev- as pefix for the branch. It will not work for us because packagist cannot handle packages in subdirectories. Maybe we should therefore extraxt the php SDK into its own repository. We could still have it in the VaaS repo as submodule.

Also. I am still around in monday, If you have any questions abiut the stuff that went here and I can read comments and chat mesages on my ride, so If you have any questions please ask.

**Comment by unglaublicherdude** [comment]
Here is where the solution comes from:
https://github.com/amphp/http-client/issues/365#issuecomment-2291491563

**Comment by unglaublicherdude** [comment]
I just wanted to add that this PR now also activates core-dumping by Apache, so we can actually investigate stuff like this. The dumps will be uploaded as artifacts in the build run.

File: .devcontainer/postCreateCommands.sh [hunk 1]
```diff
@@ -5,10 +5,14 @@ sudo apt-get install -y bash-completion vim iputils-ping telnet
 sudo bash -c "docker completion bash > /usr/share/bash-completion/completions/docker"
 sudo bash -c "composer completion bash > /usr/share/bash-completion/completions/composer"
 sudo bash -c "npm completion > /usr/share/bash-completion/completions/npm"
+sudo cp xdebug.local.ini /usr/local/etc/php/conf.d/xdebug.ini
```
Comment (unglaublicherdude) [inline .devcontainer/postCreateCommands.sh:8]: Makes local (within the devcontainer) actually work.

File: .github/workflows/release-app.yml [hunk 2]
```diff
@@ -52,26 +52,36 @@ jobs:
...
-        run: ./install.sh ${{ matrix.nextcloud_version }}
+        run: ./install.sh ${{ matrix.nextcloud_version }} 1
 
       - name: run tests
+        id: bats-tests
         env:
           CLIENT_ID: ${{ secrets.VAAS_CLIENT_ID }}
           CLIENT_SECRET: ${{ secrets.VAAS_CLIENT_SECRET }}
-        run: bats --no-parallelize-across-files --jobs 2 ./tests/bats
+        run: |
```
Comment (unglaublicherdude) [inline .github/workflows/release-app.yml:69]: This If was implemented so the artifact upload can still be done so we can investigate stuff like the coredump error.

File: Dockerfile.Nextcloud [hunk 3]
```diff
@@ -1,8 +1,36 @@
 ARG NEXTCLOUD_VERSION=29.0.4
+ARG INSTALL_XDEBUG=1
 
-FROM nextcloud:${NEXTCLOUD_VERSION}
+FROM nextcloud:29.0.4
 
-RUN apt update && apt install -y less vim telnet iputils-ping
+RUN apt update && apt install -y \
```
Comment (unglaublicherdude) [inline Dockerfile.Nextcloud:6]: Most of the changes are for helping us vie coredump files. You have to activate commented Out stuff, build the Container and then Install the /tmp/apache2-bin_* and /tmp/apache2-bin-dbgsym_* packages via apt Install ./package.deb

Then you can actually `gdb /usr/sbin/apache2 coredumpfile`

File: composer.json [hunk 4]
```diff
@@ -9,8 +9,9 @@
 		}
 	],
 	"require": {
-		"gdata/vaas": "8.0.2",
-		"coduo/php-humanizer": "^5.0"
+		"gdata/vaas": "v9.0.3",
+		"coduo/php-humanizer": "^5.0",
+		"amphp/file": "dev-opcache-bug"
```
Comment (unglaublicherdude) [inline composer.json:14]: Basically overides the dependency of the amphp/http-client package.

**Review by lennartdohmann — COMMENTED** [review]
As the topic is not currently time-critical, we can wait until we discuss it in the team in my opinion

**Comment by unglaublicherdude** [comment]
> As the topic is not currently time-critical, we can wait until we discuss it in the team in my opinion

The Change is now released with the new Version of amphp/file

File: Makefile [hunk 5]
```diff
@@ -121,6 +121,16 @@ appstore: build
 	tar czf $(appstore_package_name).tar.gz \
 	--transform s/$(app_directory_name)/$(app_real_name)/ \
 	--exclude-vcs \
+	--exclude="../$(app_directory_name)/opcache-disabled.ini" \
+	--exclude="../$(app_directory_name)/opcache-blacklist.txt" \
+	--exclude="../$(app_directory_name)/artifacts" \
+	--exclude="../$(app_directory_name)/tmp*" \
+	--exclude="../$(app_directory_name)/Dockerfile*" \
```
[the remark this answers is not in the document]
Comment (unglaublicherdude) [reply Makefile:128]: It has a wildcard. So that should already be ignored

**Review by lennartdohmann — APPROVED** [review]

**Comment by unglaublicherdude** [comment]
I rebased this PR now, it can be merged.

**Review by pstadermann — APPROVED** [review]

**Pull request merged**